packages/react-composites/src/composites/common/AvatarPersona.tsx

```diff
@@ -87,3 +82,58 @@ export const AvatarPersona = (props: AvatarPersonaProps): JSX.Element => {
     />
   );
 };
+
+/**
+ * Hook to override the avatar persona data of the users
+ * @private
+ */
+export const useDataProvider = (
+  userIds: (string | undefined)[],
+  dataProvider?: AvatarPersonaDataCallback
+): (AvatarPersonaData | undefined)[] => {
+  const [data, setData] = React.useState<(AvatarPersonaData | undefined)[]>([]);
+  useEffect(() => {
+    (async () => {
+      if (dataProvider) {
+        const newData = await Promise.all(
+          userIds.map(async (userId: string | undefined) => {
+            if (!userId) {
+              return undefined;
+            }
+            return await dataProvider(userId);
+          })
+        );
+        if (shouldUpdate(data, newData)) {
+          setData(newData);
+        }
+      }
+    })();
+  });
+  return data;
+};
+
+/**
+ * function to determine if there is new user avatar data in current render pass.
+ * @param currentData
+ * @param newData
+ */
+const shouldUpdate = (
+  currentData: (AvatarPersonaData | undefined)[],
+  newData: (AvatarPersonaData | undefined)[]
+): boolean => {
+  let newDataPresent = false;
+  if (currentData.length !== newData.length) {
+    return true;
+  }
+  currentData.forEach((p, i) => {
```

Comment:
I'd recommend extracting a helper `avatarDeepEqual(a: AvatarPersonaData, b:AvatarPersonaData): boolean`.

This can then be simplified to a oneliner: `return currentData.some((data, i) => avatarDeepEqual(data, newData[i]));`